### PR TITLE
ISPN-11390 Non-components cannot export metrics

### PR DESCRIPTION
**Pull request by anistor**
https://issues.redhat.com/browse/ISPN-11390
https://issues.redhat.com/browse/ISPN-11388
https://issues.redhat.com/browse/ISPN-11389
https://issues.redhat.com/browse/ISPN-11403
https://issues.redhat.com/browse/ISPN-11438 

Some places need manual attention because automatic registration of metrics does not work there. See the 2 mbeans in AbstractProtocolServer

### CODE REVIEW

**Comment by gustavonalle** [comment]
Tests results don't look good

**Comment by anistor** [comment]
Yes, and many on those failures are on master also. This gets me double-worried.

**Comment by anistor** [comment]
Tests should be fine now @danberindei , @gustavonalle . Some still fail, but I don't see any connection

File: server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java [hunk 1]
```diff
@@ -95,22 +95,26 @@ private int calculateGlobalConnections() {
...
-            ObjectName transportNamePattern = new ObjectName(jmxRegistration.getDomain() + ":type=Server,component=Transport,name=*");
-            Set<ObjectName> objectNames = jmxRegistration.getMBeanServer().queryNames(transportNamePattern, null);
-
-            // sum the NumberOfLocalConnections from all transport MBeans that match the pattern
-            int total = 0;
-            for (ObjectName name : objectNames) {
-               if (name.getKeyProperty("name").startsWith(serverName)) {
-                  Integer connections = (Integer) jmxRegistration.getMBeanServer().getAttribute(name, "NumberOfLocalConnections");
-                  total += connections;
+         if (jmxRegistration.enabled()) {
```
Comment (tristantarrant) [inline server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java]: I'd really love this to be rewritten not to use JMX at some point

Comment (anistor) [reply server/core/src/main/java/org/infinispan/server/core/transport/NettyTransportConnectionStats.java]: I've create https://issues.redhat.com/browse/ISPN-11466 to track this.

**Pull request merged**